lib/src/screen/dice/dice_screen.dart

```diff
@@ -30,35 +28,19 @@ class _DiceScreenState extends State<DiceScreen> {
           ),
         ),
         body: Center(
-          child: StreamBuilder(
-            stream: widget.diceBloc.diceRoll,
-            initialData: RollEvent(
-              stateType: RollStateType.initial,
-            ),
-            builder: (BuildContext context, snapshot) {
-              return Column(
-                mainAxisAlignment: MainAxisAlignment.center,
-                children: <Widget>[
-                  Text(
-                    'Roll a dice:',
-                  ),
-                  Container(
-                    height: Dimension.diceContainerSize,
-                    child: RollWidget(
-                      rollEvent: (snapshot.hasData)
-                          ? snapshot.data
-                          : RollEvent(
-                              stateType: RollStateType.loading,
-                            ),
-                    ),
-                  ),
-                ],
-              );
-            },
-          ),
-        ),
-        floatingActionButton: OnPressWidget(
-          diceBloc: widget.diceBloc,
+            child: ListView.builder(
+              shrinkWrap: true,
+              scrollDirection: Axis.horizontal,
+              itemCount: widget.diceBloc.length,
+              itemBuilder: (BuildContext context, int index) {
+                return DiceStreamBuilder(
+                  diceBloc: widget.diceBloc[index],
```

Comment:
As I mentioned before, this is not a good practice, we should have the logic for all the dices inside just one Bloc

Reply:
Done.